device/futhark/CMakeLists.txt

```diff
@@ -0,0 +1,140 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021-2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# The C language is not enabled in our project by default, but we need it to
+# compile Futhark's code.
+enable_language(C)
+
+# Try to find our dependencies, which are of course Futhark itself (required)
+# and the CUDA toolkit for compiling CUDA projects. Also notify the user of
+# what we find.
+find_program(Futhark futhark REQUIRED)
+find_package(CUDAToolkit)
+
+message(STATUS "Found Futhark compiler at ${Futhark}")
```

Comment:
:thinking: This may be a bit too much effort for this, but here we go...

I would prefer to package up most of the code in this `CMakeLists.txt` file into a `FindFuthark.cmake` module. So that it could be used conveniently throughout this project for generating code. With the familiar

```cmake
find_package(Futhark REQUIRED)
```